flink-filesystems/flink-s3-fs-base/src/main/java/org/apache/flink/fs/s3/common/writer/S3RecoverableFsDataOutputStream.java

```diff
@@ -129,10 +129,9 @@ public void sync() throws IOException {
         lock();
         try {
             fileStream.flush();
-            openNewPartIfNecessary(userDefinedMinPartSize);
+            uploadCurrentAndOpenNewPart(fileStream.getPos());
```

Comment:
We're testing the code in our internal system with file sink compaction use-case. @hlteoh37 can you plz help by executing the mentioned test?

Reply:
@MartijnVisser could you plz lend a helping hand here? That said this is working in our internal system but happy to do the homework and execute the `S3 connector test suite` but no idea what we talk about...

Reply:
@hlteoh37 @MartijnVisser please share what to execute because I've no idea about the mentioned performance test. Waiting a couple of weeks hoping somebody can reach here and can give some guidance. Otherwise I'm intended to merge it in a couple of weeks.